### ISSUE TITLE
Get list of all possible accessible methods ,class variables from a method using javaparser

### ISSUE THREAD

**Issue by sourabhsparkala** [javaparser/javaparser#947]
class A{
     public static int d;
}
class B{
    method(){
          SOP(A.d);
     }
}

In this example, is there a way to find if class B has access to the static variable 'd' in class A using the parser. Does it consider the access specifiers of the respective variables and classes? And also is it possible to find all the dependent and related classes and their accessible variables and methods from B using javaparser.

**Comment by matozoid** [comment]
In short: no. This library is a parser. It shuffles source code into a tree and nothing else.

Long answer: our side project https://github.com/javaparser/javasymbolsolver is a library that takes that tree and figures out how things are tied together. @ftomassetti knows all about it.

**Comment by ftomassetti** [comment]
You need JSS for that.

1. You can use JSS to try to resolve `d` in `A.d`: it could either be able to resolve the symbol or not. If it resolves it that means the symbol was visible and not shadowed by other symbols
2. What do you mean with related classes and their accessible variables and methods ?

**Comment by sourabhsparkala** [comment]
Thank you for the reply.

When i say accessible variables. Please find the example below.

Say 
public class A{
public static int d;
private int c;
protected int b;
}
class B extends A{
methodA(){
//List of accessible variables from methodA() would be static int d, b 
}

//Say there is another method
methodC(){
int temp;
int temp2;
//List of accessible variables from methodC() would be static int d, b, temp, temp2;
}
}

Basically, I need help in understanding if there is a way in getting the all the possible variables accessible from methodC() and methodA()

**Comment by matozoid** [comment]
You'll get better help if you move your question to https://github.com/javaparser/javasymbolsolver
